src/compiler.ts

```diff
@@ -4597,6 +4597,7 @@ export class Compiler extends DiagnosticEmitter {
           }
           this.currentType = leftType;
         }
+        this.currentFlow.inheritBranch(rightFlow);
```

Comment:
It's a little more complex than this I guess. For instance, if the LHS is always `true`, the parent flow would inherit the right flow unconditionally (not a branch), while if it is always `false`, the parent flow would not inherit the right flow since it doesn't execute.

Reply:
Yes, `inheritBranch` is a little conservative. It can spilt with difference LHS/RHS `ConditionKind`